subsys/zbus/zbus.c

```diff
@@ -44,17 +81,33 @@ int _zbus_init(void)
 SYS_INIT(_zbus_init, APPLICATION, CONFIG_ZBUS_CHANNELS_SYS_INIT_PRIORITY);
 
 static inline int _zbus_notify_observer(const struct zbus_channel *chan,
-					const struct zbus_observer *obs, k_timepoint_t end_time)
+					const struct zbus_observer *obs, k_timepoint_t end_time,
+					struct net_buf *buf)
 {
 	int err = 0;
 
-	if (obs->type == ZBUS_OBSERVER_LISTENER_TYPE) {
+	switch (obs->type) {
+	case ZBUS_OBSERVER_LISTENER_TYPE: {
 		obs->callback(chan);
-
-	} else if (obs->type == ZBUS_OBSERVER_SUBSCRIBER_TYPE) {
+	} break;
+	case ZBUS_OBSERVER_SUBSCRIBER_TYPE: {
 		err = k_msgq_put(obs->queue, &chan, sys_timepoint_timeout(end_time));
-	} else {
-		CODE_UNREACHABLE;
+	} break;
+#if defined(CONFIG_ZBUS_MSG_SUBSCRIBER)
+	case ZBUS_OBSERVER_MSG_SUBSCRIBER_TYPE: {
+		struct net_buf *cloned_buf = net_buf_clone(buf, sys_timepoint_timeout(end_time));
+
+		if (cloned_buf == NULL) {
+			err = -ENOMEM;
```

Comment:
not a fan of setting "err" mid function and hope it gets returned without anything else falling apart. Could this just `return -ENOMEM` and 112 be a `return 0`?

Reply:
Done.